Sources/Host/POSIX/FileIO.cpp

```diff
@@ -37,6 +38,29 @@ ErrorCode FileIO::CloseFile(int fd) {
   return kSuccess;
 }
 
+ErrorCode FileIO::FileRead(int fd, std::string &buf, uint64_t &count,
+                           uint64_t offset) {
+  if (offset > std::numeric_limits<off_t>::max()) {
+    return kErrorInvalidArgument;
+  }
+
+  off_t offArg = static_cast<off_t>(offset);
+  size_t countArg = static_cast<size_t>(count);
+
+  buf.resize(countArg);
+
+  ssize_t nRead = ::pread(fd, &buf[0], countArg, offArg);
```

Comment:
Ok yeah, I thought this was dumb. I'll see if I can move other analogous functions over to U8Vector.